raster-foundry/raster-foundry
# Add annotations' edit, delete, filter, clone, import, export features

## Overview

This PR adds annotations' edit, delete, filter, clone, import, export features based on updated UI design. 
* Annotations are draggable under creation, edit, and clone modes. 
* Validation on label name is set to avoid blank. Autocomplete of label names is enabled.
* Export respects the filter. 
* Fields are pre-populated if creating new under a filter or cloning.
* Shape and sidebar item highlight and autoscroll upon clicking on either are enabled. 
* Import and export are limited to geojson files.
* Fix zoom level update issue 

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

#### Import
<img width="1680" alt="import" src="https://user-images.githubusercontent.com/16109558/29220366-d4dbcce2-7e88-11e7-8ff1-93b095548247.png">

#### Create
<img width="1680" alt="create" src="https://user-images.githubusercontent.com/16109558/29220482-2a0820c6-7e89-11e7-93ae-b7e7760a09b1.png">

#### Filter and Clone
<img width="1680" alt="filter and clone" src="https://user-images.githubusercontent.com/16109558/29220522-49383aee-7e89-11e7-8c48-779436d88f26.png">

#### Edit
<img width="1680" alt="edit" src="https://user-images.githubusercontent.com/16109558/29220603-97d29316-7e89-11e7-8fda-afa119ee33ad.png">

#### Click, highlight, autoscroll
<img width="1680" alt="click and highlight" src="https://user-images.githubusercontent.com/16109558/29220619-a36c0b76-7e89-11e7-86a3-53f708f13f35.png">

### Notes
 * Uploaded geojson files must have `label` and `description` properties - this will be improved in the next round with modals.
 * Sidebar items' icons representing points and polygons, and clone buttons are placeholders. Will use real icons once we have them.

## Testing Instructions

 * Go to `projects/edit/:project_id/annotate`
 * Load your own geojson file (must have `label` and `description` properties) or start creating new annotations
 * Add, edit, delete, clone, filter, import, export to see if it behaves as you expected.

Closes #2279, #2280, #2281, #2282, #2283, #2080
